docs(content switcher): usage updates (#1411)
carbon-design-system/carbon-website: src/pages/components/content-switcher/usage.mdx

---
title: Content switcher

> description: The content switcher is used to toggle between two or more content sections within the same space on screen.

Reviewer: ```suggestion
description: Content switchers allow users to toggle between two or more content sections within the same space on screen.
```

tabs: ["Usage", "Style", "Code", "Accessibility"]
---

<PageDescription>

> The content switcher is used to toggle between two or more content sections within the same space on screen.

Reviewer: ```suggestion
Content switchers allow users to toggle between two or more content sections within the same space on screen.
```

</PageDescription>

<AnchorLinks>

<AnchorLink>Overview</AnchorLink>
<AnchorLink>Live demo</AnchorLink>
<AnchorLink>Formatting</AnchorLink>
<AnchorLink>Content</AnchorLink>
<AnchorLink>Behaviors</AnchorLink>
<AnchorLink>Related</AnchorLink>
<AnchorLink>References</AnchorLink>
<AnchorLink>Feedback</AnchorLink>

</AnchorLinks>

## Overview

> The content switcher is used to toggle between two or more content sections within the same space on screen. Only one content section is shown at a time.

Reviewer: ```suggestion
Content switchers allow users to toggle between alternate views of similar or related content. Only one content section is shown at a time.
```

### When to use

> Use a content switcher to give users alternate views of the same or similar content. Content switchers are frequently used to let users toggle between different formatting, like with a grid view and a table view. They are also often used to narrow content groups. For example, use a content switcher if you have a single category, such as “E-mail” and you want to divide it into views such as “All”, “Read”, and “Unread”.

Reviewer: ```suggestion
Content switchers are frequently used to let users toggle between different formatting, like with a grid view and a table view. They are also often used to narrow content groups. For example, use a content switcher if you have a single category, such as “E-mail” and you want to divide it into views such as “All”, “Read”, and “Unread”.
```

`}</ComponentVariant>
</ComponentDemo>

## Formatting

### Anatomy

<Row>
<Column colLg={8}>

![Anatomy of a content switcher group.](images/content-switcher-usage-2.png)

</Column>
</Row>

1. **Selected content tab:** Only one content tab can be selected at a time and there should always be one selected.
2. **Content tab:** Selectable container for each content view.
3. **Text label:** Describes the content view.

### Sizing

> The overall content switcher width varies based on the content and layout. Each individual content tab should be the same width. The content tab with the longest text label should have 16px spacing to the right of the label. All other tabs should be that width as well.

Reviewer: ```suggestion
The overall content switcher width varies based on the content and layout. Each individual content tab should be the same width. The content tab with the longest text label should have 16px spacing to the right of the label. The width of all other tabs should match the widest tab.
```

![Do base content tab width on the longest text label.](images/content-switcher-usage-3-do.png)

  </DoDont>
  <DoDont type="dont" caption="Do not use a different width for each content tab.">

![Do not use a different width for each content tab.](images/content-switcher-usage-3-don't.png)

  </DoDont>
</DoDontRow>

## Content

### Main elements

#### Text label

- Be concise and specific and limit text labels to two to three words.
- Text labels should clearly communicate the view users will see and the content contained in the view.

### Further guidance

For further content guidance, see Carbon's [content guidelines](notion://www.notion.so/guidelines/content/general).

## Behaviors

### Default selection

Only one content section can be shown at a time. The first content tab in the switcher should be determined based on usage and should always be the default selection.

### States

The content switcher includes two content tab states: **selected** and **unselected.** The selected state uses a high contrast color.

<Row>
<Column colLg={8}>

![Selected and unselected content switcher states.](images/content-switcher-usage-4.png)

</Column>
</Row>

> For detailed visual information about the various states for this component, see the Style tab.

Reviewer: Should we include a link here?

Author: Good idea. Added!
